src/azul/deployment.py

```diff
@@ -91,18 +91,15 @@ def es_endpoint(self) -> Optional[Netloc]:
 
     @property
     def es_instance_count(self) -> Optional[int]:
-        if config.es_endpoint:
-            return config.es_instance_count
+        return config.es_instance_count
 
     @property
     def lambda_env(self) -> Mapping[str, str]:
         if config.share_es_domain:
-            endpoint = "${data.aws_elasticsearch_domain.azul_es_domain.endpoint}"
-            instance_count = "${data.aws_elasticsearch_domain.azul_es_domain.cluster_config[0].instance_count}"
+            instance_count = "${var.azul_es_domain.cluster_config[0].instance_count}"
         else:
-            endpoint = "${var.azul_es_domain.endpoint}"
             instance_count = config.es_instance_count
-        return config.lambda_env((endpoint, 443), instance_count)
+        return config.lambda_env(("${var.azul_es_domain.endpoint}", 443), instance_count)
```

Comment:
Finally, if all of the parameters to `config.lambda_env` turn out to be constants, then I think the constants should be factored down into that method. Then _this_ `lambda_env` property can be removed and the one from config can always just be used instead.